### ISSUE TITLE
Make datasource imports work with ts-node

### ISSUE THREAD

**Issue by bforbis** [strongloop/loopback-next#2794]
Currently the `lb4 datasource` command will create a `{name}.datasource.ts` file and a `{name}.datasource.json` file. Since these files have the same base name, it causes issues with imports while running with `ts-node` (useful for development). By default, `ts-node` will import the JSON file over the TS file, causing datasource construction to fail. For more info on how `ts-node` works, see https://github.com/TypeStrong/ts-node/issues/769

I was able to workaround this issue by manually renaming the json file, but it would be good if the datasource generator didn't create two files with the same base name.

**Comment by dougal83** [comment]
Hey, Please excuse my ignorance but why do you use ts-node?  Curious as to the necessity, are you doing something that specifically requires it?

**Comment by bforbis** [comment]
It's not really necessary, but it makes certain things in development have a quicker feedback loop.

1. I can throw together quick ts scripts and test them with a single command
2. It allows me to skip some more complex steps of my normal build phase that involves moving static files around, as executing within ts-node will operate under the same directory structure within src/, instead of in dist/

**Comment by dougal83** [comment]
> quicker feedback loop.

Ah like build times with the watch flag?  I just use ts-node for scripts like building .env files.  Fair dos, I think this'll be low priority until lb4 has feature parity with lb3.

**Comment by bajtos** [comment]
@bforbis thank you for starting this discussion. I was looking into ts-jest and ts-node recently and discovered a similar problem: by default, LoopBack booters look for `.js` files only, and therefore don't discover `.ts` files. 

I think it would be great to improve LoopBack to support `ts-node`. Would you like to help?

For example, we can modify our CLI templates to use the following naming convention: `{name}.datasource-config.json`.

Thoughts?

/cc @strongloop/loopback-maintainers

**Comment by AaronJan** [comment]
I'm just encountering this issue too.

I used `ts-node` for virtually all my projects, naturally I changed `nom run start:dev` to `cross-env NODE_ENV ts-node src/boot-server.ts`.

But when I added a component, one of my component class injection keep getting exception message: `Cannot resolve injected arguments for XXObserver.[0]: The arguments[0] is not decorated for dependency injection, but a value is not supplied`.

Here is my code:

```typescript
import { inject, lifeCycleObserver, LifeCycleObserver } from '@loopback/core';
import { LoggerToken } from '../../bindings'; // Seems any path that is outside `__dirname` will gets `undefined`.

@lifeCycleObserver('xx')
export class XXObserver implements LifeCycleObserver {
  constructor(@inject(LoggerToken) private: logger: any) {
    //
  }
}

```

After a little digging, I find out that `LoggerToken` is `undefined` in this context (but it's not).

Therefore, in `@loopback/context/resolver`, `injectedArgs` would looks like this:

```typescript
{
  target: [Function: XXObserver],
  member: undefined,
  methodDescriptorOrParameterIndex: 0,
  bindingSelector: undefined, // No selector
  metadata: { decorator: '@inject' },
  resolve: undefined
}
```

I think this is related to this issue somehow? And looks like I can't using `ts-node` anymore, have to going old-school style..

**Comment by achrinza** [comment]
@bajtos This has probably been fixed by #4060

**Comment by bajtos** [comment]
@achrinza thank you for the comment. I am going to close this issue as resolved then.
